cms/src/symmetric_key.rs

```diff
@@ -0,0 +1,237 @@
+pub struct PSKCAlgorithmParameters<'range> {
+    pub suite: String,
+    pub challenge_format: ChallengeFormat<'range>,
+    pub response_format: ResponseFormat,
+}
+
+/// The `ChallengeFormat` type is defined in [RFC 6031 Section 3.2.7].
+///
+/// ```text
+///    ChallengeFormat ::= SEQUENCE {
+///      encoding    Encoding,
+///      checkDigit  BOOLEAN DEFAULT FALSE,
+///      min         INTEGER (0..MAX),
+///      max         INTEGER (0..MAX),
+///      ... }
+/// ```
+///
+/// [RFC 6031 Section 3.2.7]: https://datatracker.ietf.org/doc/html/rfc6031#section-3.2.7
+#[derive(Sequence, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub struct ChallengeFormat<'range> {
+    pub encoding: Encoding,
+    #[asn1(default = "Default::default")]
+    pub check_digit: bool,
+    pub min: der::asn1::IntRef<'range>,
+    pub max: der::asn1::IntRef<'range>,
+}
+
+/// The `Encoding` type is defined in [RFC 6031 Section 3.2.7].
+///
+/// ```text
+///    Encoding ::= UTF8STRING ("DECIMAL" | "HEXADECIMAL" |
+///                 "ALPHANUMERIC" |"BASE64" |"BINARY")
+/// ```
+///
+/// [RFC 6031 Section 3.2.7]: https://datatracker.ietf.org/doc/html/rfc6031#section-3.2.7
+pub type Encoding = String;
+
+/// The `ResponseFormat` type is defined in [RFC 6031 Section 3.2.7].
+///
+/// ```text
+///    ResponseFormat ::= SEQUENCE {
+///      encoding     Encoding,
+///      length       INTEGER (0..MAX),
+///      checkDigit   BOOLEAN DEFAULT FALSE,
+///      ... }
+/// ```
+///
+/// [RFC 6031 Section 3.2.7]: https://datatracker.ietf.org/doc/html/rfc6031#section-3.2.7
+#[derive(Sequence, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub struct ResponseFormat {
+    pub encoding: Encoding,
+    pub length: u32,
+    #[asn1(default = "Default::default")]
+    pub check_digit: bool,
+}
+
+/// The `ValueMac` type is defined in [RFC 6031 Section 3.2.12].
+///
+/// ```text
+///    ValueMac ::= SEQUENCE {
+///      macAlgorithm UTF8String,
+///      mac          UTF8String }
+/// ```
+///
+/// [RFC 6031 Section 3.2.12]: https://datatracker.ietf.org/doc/html/rfc6031#section-3.2.12
+#[derive(Sequence, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub struct ValueMac {
+    pub mac_algorithm: String,
+    pub mac: String,
+}
+
+/// The `PSKCKeyUsages` type is defined in [RFC 6031 Section 3.3.4].
+///
+/// ```text
+///    PSKCKeyUsages ::= SEQUENCE OF PSKCKeyUsage
+/// ```
+///
+/// [RFC 6031 Section 3.3.4]: https://datatracker.ietf.org/doc/html/rfc6031#section-3.3.4
+pub type PSKCKeyUsages<'keys_usage> = Vec<PSKCKeyUsage<'keys_usage>>;
+
+/// The `PSKCKeyUsage` type is defined in [RFC 6031 Section 3.3.4].
+///
+/// ```text
+///    PSKCKeyUsage ::= UTF8String ("OTP" | "CR" | "Encrypt" |
+///                     "Integrity" | "Verify" | "Unlock" | "Decrypt" |
+///                     "KeyWrap" | "Unwrap" | "Derive" | "Generate")
+/// ```
+///
+/// [RFC 6031 Section 3.3.4]: https://datatracker.ietf.org/doc/html/rfc6031#section-3.3.4
+pub type PSKCKeyUsage<'pskc_key_usage> = String;
```

Comment:
the lifetime isn't used.
```suggestion
pub type PSKCKeyUsage = String;
```